Added interchromosome interaction data

kajocina opened the pull request:
Tested the exported dataframes with raw HiC matrices in JuiceBox and it looks fine.

kajocina committed:
seems to work but needs testing

kajocina committed:
added one import

kajocina committed:
only INTER- normalization vectors supported

kajocina committed:
roxygenized

kajocina committed:
removed a test line

liz-is commented:
Thanks for the pull request! :)

I tried to test your code on the mouse data, but it stops because the interchromosomal normalisation files are not present. I think it should be possible to read in the data using the intrachromosomal normalisation vectors, if the *inter*chromosomal ones are not available. After all, the interchromosomal normalisation is only calculated for human and only at lower resolutions. Could you add this functionality?

Also a few minor things:

* the `read_rao_huntley_data_interchr` function is not exported
* a couple of things come up in R CMD check:
> * checking package dependencies ... ERROR
Namespace dependency not required: 'gtools'

> checking Rd \usage sections ... WARNING
Undocumented arguments in documentation object 'make_is_from_data'
  ‘interchr’
Undocumented arguments in documentation object 'read_rao_huntley_data'
  ‘read_expected’

If you can fix these I'm happy to merge the pull request.

kajocina committed:
fixed bin numbers in anchor2

kajocina commented:
Hi!

Do you know by any chance why the normalization vectors have different length in interchromosomal interaction folders? The KRnorm has the same length as for intrachromosomal normalization. It's the inter-KR that is 3 bins short...

E.g. 

`2493 lines in chr1_100kb.INTERKRnorm`
`2496 lines in  chr1_100kb.KRnorm`

Best,
Piotr

kajocina committed:
added normalization, single KR/VC not yet

Reverted the fix with bin numbers, it wasn't working right. The only problem now that if users access data with interactions(iset)@anchor2, they will get numbers pointing to the position in the dataframe, but not the real bin of the chromosome.

liz-is commented:
I have no idea, sorry! I can't download the data at the moment to explore it, either.

kajocina commented:
Ok I will try checking later, so far I added only genome-wide normalization
but KR/VC need to wait since I can't add them easily into the dataframe in
the same way...

On Thu, Apr 13, 2017 at 3:32 PM, Liz Ing-Simmons <notifications@github.com>
wrote:

> I have no idea, sorry! I can't download the data at the moment to explore
> it, either.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/liz-is/readhic/pull/6#issuecomment-293896315>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADyH3AgyO-i2VTm8-V6RLAcUon4vcd1aks5rviP8gaJpZM4MrO5v>
> .
>

liz-is commented:
Which dataset are you using, that has the interchromosomal normalisation vectors calculated? Is it only the GM12878 combined dataset that has these available? I wanted to download some of the data to look into this, but I'm not sure if I need to download that dataset or one of the smaller ones will do... If it is the combined dataset that you're using, I don't suppose you could email me a subset of the data for testing purposes? Just a couple of the matrices and vectors at 1MB resolution will do. I'm reluctant to download the whole thing because the machines I have access to at the moment have limited storage available (!). If you can, my gmail address is liz.ingsimmons.

Thanks!

kajocina committed:
Update README.md

kajocina committed:
Update README.md

kajocina committed:
Update README.md
